controllers/appwrapper_controller.go

```diff
@@ -95,8 +79,14 @@ const (
 func (r *AppWrapperReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 
 	_ = log.FromContext(ctx)
-
+	// todo: Move the getOCMClusterID call out of reconcile loop.
+	// Only reason we are calling it here is that the client is not able to make
+	// calls until it is started, so SetupWithManager is not working.
+	if !useMachineSets && ocmClusterID == "" {
+		getOCMClusterID(r)
```

Comment:
I'd removed the call here, and get the cluster ID lazily, by calling the function where it's needed, and gate the retrieval of the ID with `sync.Once`.

Reply:
I tried to implement this by calling it within the `machinePoolExists()` function in `SetupWithManager()` but this didn't work because the client can't be used until it is started. 

I do notice that we check if the [ocm secret exists](https://github.com/project-codeflare/instascale/blob/main/controllers/appwrapper_controller.go#L170) before we call `machinePoolExists()`. As I understand it we only need that ocm secret if we are scaling machine pools. If this is the case I can use that as the condition that sets `useMachinePools` to `true` and apply your suggestion in the other machinePool functions.

Reply:
So the call from the `SetupWithManager` method fails because `getOCMClusterID` uses the controller-runtime cached client to query the ClusterVersion API. This is not recommended as it'll keep watching the API for no reason, and should be changed to using the plain client-go / openshift client. Then it'll be possible to move the call to `getOCMClusterID` in `SetupWithManager`.